src/components/structures/login/Registration.js

```diff
@@ -209,24 +210,43 @@ module.exports = React.createClass({
                 accessToken: response.access_token
             });
 
-            // Auto-join rooms
-            if (self.state.teamsConfig && self.state.teamsConfig.teams) {
-                for (let i = 0; i < self.state.teamsConfig.teams.length; i++) {
-                    let team = self.state.teamsConfig.teams[i];
-                    if (self.state.formVals.email.endsWith(team.domain)) {
-                        console.log("User successfully registered with team " + team.name);
+            if (
+                self._rtsClient &&
+                self.props.queryParams &&
+                self.props.queryParams.referrer &&
+                self.state.teamSelected
+            ) {
+                // Track referral, get team_token in order to retrieve team config
+                self._rtsClient.trackReferral(
+                    self.props.queryParams.referrer,
+                    response.user_id,
+                    self.state.formVals.email
+                ).then((args) => {
+                    const teamToken = args[1].team_token;
+                    // Store for use /w welcome pages
+                    window.localStorage.setItem('mx_team_token', teamToken);
```

Comment:
I hadn't considered what happens when `mx_team_token` gets removed from local storage. Riot would not consider them part of the team at that point. We might need another endpoint on the RTS to be hit on login to re-verify team membership and send the `team_token` again.